### PR TITLE
Соловьев - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by qwert182**

### CODE REVIEW

**Comment by qwert182** [comment]
@qwert183, @ipet3o please review.

**Comment by qwert183** [comment]
Замечаний нет :+1:

File: code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    public static Fraction subtract(final Fraction a, final Fraction b) {
+        int lcm = lcm(a.denominator, b.denominator);
+        return new Fraction(
+                a.numerator * (lcm / a.denominator) - b.numerator * (lcm / b.denominator),
+                lcm
+        );
+    }
+
```
Comment (ipet3o) [inline code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java]: Remove extra new lanes

File: code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    @Override
+    public int hashCode() {
+        return numerator ^ denominator;
+    }
+
+    @Override
+    public String toString() {
+        return String.valueOf(numerator) + '/' + String.valueOf(denominator);
+    }
+
```
Comment (ipet3o) [inline code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java]: Remove extra new lanes

File: code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            denominator = -denominator;
+        }
+    }
+
+    private void reduce() {
+        int d = gcd(abs(numerator), denominator);
+        numerator /= d;
+        denominator /= d;
+    }
+
```
Comment (ipet3o) [inline code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java]: Same comment

File: code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+    private static int gcd(final int number1, final int number2) {
```
Comment (ipet3o) [inline code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java]: It is not clear from the name of the function that the function does.

File: code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+
+
+
+
+
+    private void correctSign() {
+        if (denominator < 0) {
+            numerator = -numerator;
```
Comment (ipet3o) [inline code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java]: Test code never invoked this code

File: code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        return numerator ^ denominator;
+    }
+
+    @Override
+    public String toString() {
+        return String.valueOf(numerator) + '/' + String.valueOf(denominator);
```
Comment (ipet3o) [inline code/solovev-dmitriy/src/main/java/ru/unn/agile/Fraction/Model/Fraction.java]: Test code never invoked this code

**Comment by qwert182** [comment]
@kirill-korniakov

**Pull request merged**